### DISCUSSION TITLE
CLI - C what you Like

### DISCUSSION THREAD

**Discussion by aFuzzyBear** [withastro/roadmap#144] [Proposal]
Hey guys, Im looking to start in the next couple of days, 🤞  if not by the weekend at looking at the CLI tool for Astro, and giving it a nice wee tart up. I have a couple of ideas that I would personally like to see in the CLI too, these just being funkier interactions, colours, and clearer logging messages.

I have had a couple of projects that got me playing int ht ecli whilst I was learning node, and well Id like to do that here,

let me know of what you own idea's, thoughts and wishes are and we'll see what this bear can conjure up,

**Comment by FredKSchott** [comment]
Have fun! One thing that could be nice would be a centralized place for error messages. This is a pattern that I've seen in other advanced CLIs (Yarn comes to mind) that sets the stage to add support for i18n translations.

```js
// Go from this:
warn(options.logging, 'config', `Set "buildOptions.site" to generate correct canonical URLs and sitemap`);

// to this:
warn(options.logging, 'config', msg.buildOptionsSiteMissing());
```

```js
// Go from this:
warn(this.logging, 'routeCache', `Internal Warning: route cache overwritten. (${route.component})`);

// To this:
warn(this.logging, 'routeCache', msg.routeCacheOverwrittenWarning(route.component));
```
